src/datepicker/datepicker-input.ts

```diff
@@ -373,7 +382,16 @@ export class NgbInputDatepicker implements OnChanges,
       this._changeDetector.markForCheck();
 
       // restore focus
-      const elementToFocus = this._elWithFocus && this._elWithFocus['focus'] ? this._elWithFocus : this._document.body;
+      let elementToFocus = isFocusable(this._elWithFocus) ? this._elWithFocus : this._document.body;
+      if (isString(this.restoreFocus)) {
+        const testElement = this._document.querySelector(this.restoreFocus);
+        if (isFocusable(testElement)) {
+          elementToFocus = testElement;
+        }
+      } else if (isFocusable(this.restoreFocus)) {
+        elementToFocus = this.restoreFocus;
+      }
+
       elementToFocus.focus();
     }
```

Comment:
Usage of isFocusable is problably not useful here, as we know that html elements are handled (just a check for a not null should be enough)

Reply:
Thanks, it makes sense. Will remove `isFocusable()` as discussed